### PR TITLE
DOMAttr: setting value triggers XML entity parsing

### PR DESCRIPTION
**Pull request by DennisBirkholz**
See https://3v4l.org/Lhu2c

### CODE REVIEW

File: reference/dom/domattr.xml [hunk 1]
```diff
@@ -122,6 +122,13 @@
      <term><varname>value</varname></term>
      <listitem>
       <para>The value of the attribute</para>
+      
+      <para>
+       Note, XML entities are decoded upon setting value.
```
Comment (cmb69) [inline reference/dom/domattr.xml:127]: I think they are encoded (not decoded), but the proper term seems to be expanded.

File: reference/dom/domattr.xml [hunk 2]
```diff
@@ -122,6 +122,13 @@
      <term><varname>value</varname></term>
      <listitem>
       <para>The value of the attribute</para>
+      
+      <para>
```
Comment (cmb69) [inline reference/dom/domattr.xml:126]: I suggest to turn that into a note (i.e. wrap the new `<para>` in a `<note>` element). Or maybe it should even be a `<caution>`.

File: reference/dom/domattr.xml [hunk 3]
```diff
@@ -122,6 +122,13 @@
...
+       Note, XML entities are decoded upon setting value.
+       Thus the <constant>&amp;</constant> character has a special meaning.
+       Setting <varname>value</varname> to itself will fail when <varname>value</varname> contains an <constant>&amp;</constant>.
+       Use <function>DOMElement::setAttribute</function> to set unescaped strings instead.
```
Comment (cmb69) [inline reference/dom/domattr.xml:130]:
```suggestion
       To avoid entity expansion, use <methodname>DOMElement::setAttribute</methodname> instead.
```

**Review by cmb69 — COMMENTED** [review]
Thank you for the PR (which apparently felt through the cracks)! It is actually missing documentation regarding https://github.com/php/php-src/pull/10132, and given that the current behavior doesn't match the DOM specification, I think we should make it more prominent. And maybe we should add an example.

**Comment by Girgias** [comment]
This needs a rebase.

Also @nielsdos does this description make sense to you?

**Comment by nielsdos** [comment]
If the changes requested by @cmb69 are applied, and this PR is rebased, then it's good.

**Comment by DennisBirkholz** [comment]
I rebased the patch and added the changes proposed by @cmb69.

**Comment by Girgias** [comment]
Thank you!

**Pull request merged**